### PR TITLE
Suppress number of files generated to be the number of max_threads

### PR DESCRIPTION
**Pull request by sonots**
Suppress number of files generated to be max_threads in LocalExecutor even if number of input tasks are larger than max_threads.

PURPOSE:

Think of using FileInput plugins such as input-file, input-s3, and there are a lots of files as inputs. Before, embulk-output-bigquery generated local files of the same number with input files, and tried to upload them in parallel. 

Since we dropped `max_load_parallels` option as a request of frsyuki at https://github.com/embulk/embulk-output-bigquery/pull/35, number of concurrency to upload files can not be suppressed, which results in consuming lots of network resources. Futhermore, loading lots of small files into bigquery is not efficient, it is better to merge them beforehand. 

This feature suppresses number of files to be generated to be same with max_threads, which results in much more efficient.

### CODE REVIEW

**Comment by sonots** [comment]
@civitaspo could you try this with our production environment?

File: lib/embulk/output/bigquery/file_writer.rb [hunk 1]
```diff
@@ -31,20 +31,35 @@ def initialize(task, schema, index, converters = nil)
...
-            Embulk.logger.warn { "embulk-output-bigquery: unlink already existing #{@path}" }
-            File.unlink(@path) rescue nil
+          @io = create_io
+          @path = @io.path
+        end
+
+        THREAD_LOCAL_IO_KEY = :embulk_output_bigquery_file_writer_io
+
+        # Create one io object for one thread, that is, share among tasks
+        # Close theses shared io objects in transaction
```
Comment (sonots) [inline lib/embulk/output/bigquery/file_writer.rb:48]: I am requesting to add new API to get the number of `max_threads` at https://github.com/embulk/embulk/issues/412, but currently it is not available (in my understanging).

That is why I am creating new io objects **dynamically** using the thread local object to get max_threads number of io objects at here.

**Comment by sonots** [comment]
@civitaspo could you try?

**Comment by civitaspo** [comment]
@sonots Sorry for the delay of responce...
I tried the branch, but this doesn't work, because the temporary gzip files are cut off in the middle of writing.
Could you investigate this? (I couldn't find the cause in the codes.)

**Comment by sonots** [comment]
We've confirmed this worked well

**Comment by sonots** [comment]
v0.3.0.pre3 is released. Next release would become v0.3.0.

**Comment by civitaspo** [comment]
thx @sonots :+1:

**Pull request merged**